File: solutions/0001-two-sum/solution.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # Dictionary to map the number to its index
        seen = {}
        
        for index, num in enumerate(nums):
            # Calculate the required pair value
            complement = target - num
            
            # Check if the complement has already been seen
            if complement in seen:
                return [seen[complement], index]
            
            # Store the current number and its index
            seen[num] = index
            
        return []

File: solutions/0001-two-sum/solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        // Map to store the value and its corresponding index
        Map<Integer, Integer> map = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            
            // If the complement exists in the map, we found the pair
            if (map.containsKey(complement)) {
                return new int[] { map.get(complement), i };
            }
            
            // Otherwise, store the current number and its index
            map.put(nums[i], i);
        }
        
        // Return an empty array or throw an exception if no solution is found
        throw new IllegalArgumentException("No two sum solution");
    }
}

